test-utils/runtime/src/substrate_test_pallet.rs

```diff
@@ -0,0 +1,292 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{AccountId, AuthorityId, BlockNumber, Runtime, Signature, Transfer};
+use codec::KeyedVec;
+use frame_support::storage;
+use sp_core::storage::well_known_keys;
+use sp_io::hashing::blake2_256;
+use sp_std::prelude::*;
+
+const NONCE_OF: &[u8] = b"nonce:";
+const BALANCE_OF: &[u8] = b"balance:";
+
+pub use self::pallet::*;
+
+const LOG_TARGET: &str = "substrate_test_pallet";
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {}
+
+	// The current block number being processed. Set by `on_initialize`.
+	#[pallet::storage]
+	pub type Number<T: Config> = StorageValue<_, T::BlockNumber, OptionQuery>;
+
+	#[pallet::storage]
+	pub type NewAuthorities<T> = StorageValue<_, Vec<AuthorityId>, OptionQuery>;
+
+	#[pallet::storage]
+	pub type Authorities<T> = StorageValue<_, Vec<AuthorityId>, ValueQuery>;
+
+	#[pallet::genesis_config]
+	#[cfg_attr(feature = "std", derive(Default))]
+	pub struct GenesisConfig {
+		pub authorities: Vec<AuthorityId>,
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig {
+		fn build(&self) {
+			<Authorities<T>>::put(self.authorities.clone());
+		}
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(n: T::BlockNumber) -> Weight {
+			Number::<T>::put(n);
+			Weight::zero()
+		}
+
+		fn on_finalize(_n: T::BlockNumber) {}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight(100)]
+		pub fn authorities_change(
+			origin: OriginFor<T>,
+			new_authorities: Vec<AuthorityId>,
+		) -> DispatchResult {
+			frame_system::ensure_signed(origin)?;
+			<NewAuthorities<Runtime>>::put(new_authorities.to_vec());
+			Ok(())
+		}
```

Comment:
This seems to be used by `push_authorities_change_block` in `sc-network-test`. `push_authorities_change_block` itself is not being called anywhere, so `push_authorities_change_block` and this call here can be removed.

Reply:
Done.